Persuasive subgenre annotations

hmolloy opened the pull request:
In the interest of incorporating data and documentation on the Subgenre Annotation completed this past spring, I have edited/added the following documents:

1.  **writing_features.csv**: This features file now contains persuasive subgenre classifications and descriptors in the final two columns (DP and DQ). Column DP now contains data concerning the `Persuasive_Subgenre_Classification` and Column DQ contains data concerning the corresponding `Persuasive_Subgenre_Description`.
2.  **docs/writing_features_columns.csv**: This documentation now contains new rows (121-122) describing the new subgenre annotation columns added to the `writing_features.csv` file: `Persuasive_Subgenre_Classification` and `Persuasive_Subgenre_Description`.
3. **docs/forms/persuasive_subgenre_annotation.pdf**: This is a new PDF describing the subgenre annotation procedures.
4. **ReadMe**: The "Additional Documents" section now contains a new bullet introducing the `persuasive_subgenre_annotation.pdf` file.

hmolloy committed:
Persuasive Subgenre Annotations

Writing_features.csv now includes persuasive subgenre annotations.  These are located in two new columns (DP and DQ).  Column DP contains the "Persuasive Subgenre Classification" and column DQ contains the "Persuasive Subgenre Description."

hmolloy committed:
Persuasive Subgenre Annotation Descriptors

This file now contains an explanation of the two new additions to the writing_features.csv file: column DP ("Persuasive_Subgenre_Classification") and column DQ ("Persuasive_Subgenre_Description").

hmolloy committed:
Persuasive Subgenre Annotation

This document describes the taxonomy and procedures for the persuasive subgenre annotation.

hmolloy committed:
Additional Docs > Subgenre Analysis

Includes a description of the newest addition to our documentation, "persuasive_subgenre_annotation.pdf."

desilinguist commented on README.md:
```diff
@@ -45,6 +45,8 @@ The following additional files can be found under [`docs`](docs).
 
 - [`processes/genre_annotation.pdf`](docs/processes/genre_annotation.pdf) : This file contains a description of the genre annotation performed to classify writing assignments based on broad assignment types.
 
+- [`persuasive_subgenre_annotation.pdf`](docs/processes/persuasive_subgenre_annotation.pdf) : This file contains a description of the subgenre annotation performed to classify persuasive writing assignments into one of six finer-grained categories based on argument value (on a continuum of low to high), source use and integration, and support. 
```
```suggestion
- [`processes/persuasive_subgenre_annotation.pdf`](docs/processes/persuasive_subgenre_annotation.pdf) : This file contains a description of the subgenre annotation performed to classify persuasive writing assignments into one of six finer-grained categories based on argument value (on a continuum of low to high), source use and integration, and support. 
```

desilinguist commented on docs/writing_features_columns.csv:
```diff
@@ -1,160 +1,160 @@
 ? Criminology 
 ? English
 ? History
-? Sociology",,,,
```
Hmm, I am a little concerned about these commas being deleted. Do we no longer have the extra columns?

hmolloy replied:
Hi @desilinguist! I will admit that I don't actually see those commas in the penultimate version of this file, so I'm not sure what change is being detected here.  Any thoughts on what it might be?  I only intended to make edits to the last two rows of this file (rows 121 and 122).  Do you want me to create a new pull request just for the writing_features_columns.xlsx file and make the intended edits again?
![image](https://user-images.githubusercontent.com/29762455/94473759-a2a13f80-019a-11eb-99c0-146cfe6b25a6.png)

desilinguist replied:
Hmm, I am not sure. Let me try to read in the file in `pandas` and see if it works without the commas.

desilinguist replied:
@hmolloy just to confirm there are supposed to be 121 columns in the new version of the CSV file correct?

Also, I think instead of using strings like "Not applicable or not available" for the `Persuasive_Subgenre_Description` column, we should use something like "N/A" since `pandas` will properly recognize that as not-a-number rather than a custom string. If you need to explain what "N/A" means, you can do so in the PDF.

hmolloy replied:
@desilinguist there are supposed to be 121 columns in the `writing_features.csv` file and 3 columns in the writing_features_columns.csv file (plus 122 rows).

I'm fine with changing `Not applicable or not available` to "N/A" for the `Persuasive_Subgenre_Description` column.  Shall I submit a new pull request?

desilinguist replied:
Hmm, when I try to load `writing_feature_columns.csv` , I get a unicode error. 

```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x96 in position 232: invalid start byte
```

Please make sure that you are setting the encoding to UTF-8 when exporting the CSV file.

As for the "N/A" stuff, you can just fix them in this PR itself.

hmolloy replied:
Hi @desilinguist.  I've made the following changes and committed them to the `Persuasive subgenre annotations` branch, but I'm not sure if those edits are still associated with this pull request.   

1. I saved those two .csv files with UTF-8 encoding and re-uploaded them.
2. I also changed `Not applicable or not available` to "`N/A`" for the `Persuasive_Subgenre_Description` column.

Please let me know how to proceed.

Thanks so much!

desilinguist reviewed (changes requested):
Looks pretty good @hmolloy! I have a suggestion and a question.

hmolloy committed:
Update README.md

Language edits from Nitin ("source use integration, and support" changed to "source use and integration, and support")

Co-authored-by: Nitin Madnani <nmadnani@ets.org>

hmolloy committed:
Changed encoding of csv files to UTF-8. Changed 

(1) Re-saved writing_features.csv and writing_features_columns with UTF-8 encoding.
(2) Changed the string "Not applicable or not available" to "N/A" for the Persuasive_Subgenre_Description column in writing_features.csv.

hmolloy committed:
Deduplication of writing_features_columns.csv

I moved the new version of writing_features_columns.csv (now saved with UTF-8 encoding) to docs\writing_features_columns.csv, where it belongs.  The new copy with UTF-8 encoding replaced the old copy from a week ago.

hmolloy commented:
Hi @desilinguist!  I think the correct `writing_features_columns.csv` file should now be in `docs\` instead of in the parent directory.  Let me know if there's anything else you need!

desilinguist reviewed (approved):
Looks great! Thanks for all the changes!
